src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const CloneObject = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(CloneObject, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete CloneObject[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in CloneObject) {
+          delete CloneObject[key];
+        }
+        break;
+    }
+
+    result.push({ ...CloneObject });
```

Comment:
It's better to push a copy of the object to the 'result' array at the end of the loop cycle, but outside of the 'switch' block, after the current action is processed. This ensures that the same action is not repeated for each case.